Tasks/AppCenterDistributeV1/appcenterdistribute.ts

```diff
@@ -88,69 +105,157 @@ function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: stri
     request.post({ url: beginUploadUrl, headers: headers }, (err, res, body) => {
         responseHandler(defer, err, res, body, () => {
             let response = JSON.parse(body);
-            let uploadInfo: UploadInfo = {
-                upload_id: response['upload_id'],
-                upload_url: response['upload_url']
+            if (!response.package_asset_id || (response.statusCode && (response.statusCode < 200 || response.statusCode >= 300))) {
+                defer.reject(`failed to create release upload. ${response.message}`)
             }
 
-            defer.resolve(uploadInfo);
+            defer.resolve(response);
         });
     });
 
     return defer.promise;
 }
 
-function uploadRelease(uploadUrl: string, file: string, userAgent: string): Q.Promise<void> {
+function loadReleaseIdUntilSuccess(apiServer: string, apiVersion: string, appSlug: string, uploadId: string, token: string, userAgent: string): Q.Promise<string> {
+    let defer = Q.defer<string>();
+    const timerId = setInterval(async () => {
+        let response;
+        try {
+            response = await getReleaseId(apiServer, apiVersion, appSlug, uploadId, token, userAgent);
```

Comment:
Could setTimeout be better here? For the cases when request takes more time when defined as timeout - so it will be resent only after the response?

Reply:
Not exactly sure what you mean here

Reply:
Current logic sents request periodically, but not sure that it's necessary here... Could you just sent it once and wait for response?

Reply:
`setTimeout` is not applicable here, we need to periodically ask the status. This is not long running one operation.

Reply:
No, it's not long polling API

Reply:
Just still don't understand why setInterval is required...
As I see for any case - after response will be received - there will be clean up for interval
Why do we need to call it periodically?
![image](https://user-images.githubusercontent.com/8779682/87534936-62541a00-c69f-11ea-94c2-9f0b4064438a.png)